pennylane/templates/tensornetworks/ttn.py

```diff
@@ -210,18 +210,18 @@ def compute_decomposition(
         Returns:
             list[.Operator]: decomposition of the operator
         """
-        op_list = []
-        if block.__code__.co_argcount > 2:
-            for idx, w in enumerate(ind_gates):
-                op_list.append(block(*weights[idx], wires=w))
-        elif block.__code__.co_argcount == 2:
-            for idx, w in enumerate(ind_gates):
-                op_list.append(block(weights[idx], wires=w))
-        else:
-            for idx, w in enumerate(ind_gates):
-                op_list.append(block(wires=w))
 
-        return op_list
+        with qml.tape.QuantumTape(do_queue=False) as tape:
```

Comment:
Is there a reason using a tape would be preferred here? 

Queueing things into `QuantumTape` requires a global lock and thus is incompatible with any type of parallelization. Also, it performs a bunch of validation and overhead we don't need in this situation.

I'm generally not fan of using `QuantumTape` objects when they are not needed, but that might just be my opinion.

Reply:
Hi @albi3ro, thanks for the review! I don't like `QuantumTape` either but the problem is `block` is just a quantum function that may return nothing. e.g. the following is an example of `block`
https://github.com/PennyLaneAI/pennylane/blob/3ac20c631692910577ff12cd0c02bd5bc955cae5/tests/templates/test_tensornetworks/test_TTN.py#L34-L36
I also hope to have a lightweight method we can use in this situation...